### PR TITLE
VIDEO-10366 Update docs

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### Description

- Mentioning some possible causes of the warnings in the API docs
- Removing public references of TwilioWarning as this created confusion to some folks. See [thread](https://twilio.slack.com/archives/C031MM5GY69/p1654021970923649). TwilioWarning will mean more in the future so removing it for now.

### Before review
* [x] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [x] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [x] Ready for review

### CODE REVIEW

File: tsdef/types.d.ts [hunk 1]
```diff
@@ -98,8 +98,6 @@ export type BandwidthProfileMode = 'grid' | 'collaboration' | 'presentation';
 export type VideoContentPreferencesMode = 'auto' | 'manual';
 export type ClientTrackSwitchOffControl = 'auto' | 'manual';
 
-export type TwilioWarning = 'recording-media-lost';
```
Comment (makarandp0) [inline tsdef/types.d.ts:101]: I think its good to have the specific type for this. Maybe we can rename it to `NotifyWarning`

File: tsdef/types.d.ts [hunk 2]
```diff
@@ -190,7 +188,7 @@ export interface ConnectOptions {
   maxVideoBitrate?: number | null;
   name?: string | null;
   networkQuality?: boolean | NetworkQualityConfiguration;
-  notifyWarnings?: Array<TwilioWarning>;
+  notifyWarnings?: Array<'recording-media-lost'>;
```
Comment (makarandp0) [inline tsdef/types.d.ts:191]: Even though we are not using the term TwilioWarning, having a type to identify possible warnings would be good. How about calling them NotifyWarnings?

```
export type NotifyWarnings = 'recording-media-lost';
```

**Review by makarandp0 — APPROVED** [review]
Have a minor suggestion, but LGTM

**Comment by charliesantos** [comment]
Thanks for the review @makarandp0 . I pushed some minor fixes related to our ts definitions.

**Pull request merged**